samplers/probability/consistent/sampler.go

```diff
@@ -0,0 +1,178 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consistent // import "go.opentelemetry.io/contrib/samplers/probability/consistent"
+
+import (
+	"fmt"
+	"math/bits"
+	"math/rand"
+	"sync"
+
+	"go.opentelemetry.io/otel"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type (
+	ProbabilityBasedOption interface {
+		apply(*consistentProbabilityBasedConfig)
+	}
+
+	consistentProbabilityBasedConfig struct {
+		source rand.Source
+	}
+
+	consistentProbabilityBasedRandomSource struct {
+		rand.Source
+	}
+
+	consistentProbabilityBased struct {
+		// "LAC" is an abbreviation for the logarithm of
+		// adjusted count.  Greater values have greater
+		// representivity, therefore lesser sampling
+		// probability.
+
+		// lowLAC is the lower-probability log-adjusted count
+		lowLAC uint8
+		// highLAC is the higher-probability log-adjusted
+		// count.  except for the zero probability special
+		// case, highLAC == lowLAC - 1.
+		highLAC uint8
+		// lowProb is the probability that lowLAC should be used,
+		// in the interval (0, 1].  For exact powers of two and the
+		// special case of 0 probability, lowProb == 1.
+		lowProb float64
+
+		// lock protects rnd
+		lock sync.Mutex
+		rnd  *rand.Rand
+	}
+)
+
+// WithRandomSource sets the source of the random number used in this
+// prototype of OTEP 170, which assumes the randomness is not derived
+// from the TraceID.
+func WithRandomSource(source rand.Source) ProbabilityBasedOption {
+	return consistentProbabilityBasedRandomSource{source}
+}
+
+func (s consistentProbabilityBasedRandomSource) apply(cfg *consistentProbabilityBasedConfig) {
+	cfg.source = s.Source
+}
+
+// ProbabilityBased samples a given fraction of traces.  Based on the
+// OpenTelemetry specification, this Sampler supports only power-of-two
+// fractions.  When the input fraction is not a power of two, it will
+// be rounded down.
+// - Fractions >= 1 will always sample.
+// - Fractions < 2^-62 are treated as zero.
+//
+// This Sampler sets the `sampler.adjusted_count` attribute.
```

Comment:
The `adjusted_count` seems to have been removed, no?

Reply:
```suggestion
// This Sampler sets the OpenTelemetry tracestate p-value and/or r-value.
```